java/client/src/main/java/com/youtube/vitess/client/StreamIterator.java

```diff
@@ -0,0 +1,37 @@
+package com.youtube.vitess.client;
+
+import java.util.NoSuchElementException;
+
+/**
+ * StreamIterator is used to access the results of a streaming call.
+ *
+ * It is similar to java.util.Iterator, but the hasNext() method is
+ * understood to block until either a result is ready, an error occurs,
+ * or there are no more results. Also, unlike Iterator, these methods
+ * can throw VitessException or VitessRpcException.
+ */
+public interface StreamIterator<E> {
```

Comment:
Maybe we could just re-use Java's Iterator interface instead of creating a new one ?